pilot/cmd/istioctl/inject.go

```diff
@@ -58,6 +59,34 @@ func getMeshConfigFromConfigMap(kubeconfig string) (*meshconfig.MeshConfig, erro
 	return model.ApplyMeshConfigDefaults(yaml)
 }
 
+func getInjectConfigFromConfigMap(kubeconfig string) (string, error) {
+	_, client, err := kube.CreateInterface(kubeconfig)
+	if err != nil {
+		return "", err
+	}
+	config, err := client.CoreV1().ConfigMaps(istioNamespace).Get(injectConfigMapName, metav1.GetOptions{})
+	if err != nil {
+		log.Infof("could not find valid configmap %q from namespace  %q: %v - "+
+			"Use --injectConfigFile or re-run kube-inject with `-i <istioSystemNamespace> and ensure istio-inject configmap exists",
+			injectConfigMapName, istioNamespace, err)
+		return "", nil
+	}
+	// values in the data are strings, while proto might use a
+	// different data type.  therefore, we have to get a value by a
+	// key
+	injectData, exists := config.Data[injectConfigMapKey]
+	if !exists {
+		return "", fmt.Errorf("missing configuration map key %q", injectConfigMapKey)
+	}
+	log.Debugf("read inject template from configmap.")
+	var injectConfig inject.Config
+	if err := yaml.Unmarshal([]byte(injectData), &injectConfig); err != nil {
+		return "", err
+	}
+	sidecarTemplate := injectConfig.Template
+	return sidecarTemplate, nil
```

Comment:
return injectConfig.Template, nil (i.e. remove the assignment of sidecarTemplate)